test/internal/connection-providers.test.js

```diff
@@ -137,7 +138,7 @@ describe('LoadBalancer', () => {
   it('initializes routing table with the given router', () => {
     const connectionPool = newPool();
     const loadBalancingStrategy = new LeastConnectedLoadBalancingStrategy(connectionPool);
-    const loadBalancer = new LoadBalancer('server-ABC', {}, connectionPool, loadBalancingStrategy, NO_OP_DRIVER_CALLBACK);
+    const loadBalancer = new LoadBalancer('server-ABC', {}, connectionPool, loadBalancingStrategy, NO_OP_DRIVER_CALLBACK, Logger.noOp());
```

Comment:
What if the logger argument isn't defined, does it break?

Reply:
Yes, `LoadBalancer` tests in connection-providers.test.js break. This is expected, right?

Reply:
Maybe. Or maybe a noop function could be the default value. 
Otoh, these are internal stuff that can have breaking changes, so I guess this is totally fine.